apps/android/app/src/main/java/chat/simplex/app/SimplexApp.kt

```diff
@@ -65,6 +64,9 @@ class SimplexApp: Application(), LifecycleEventObserver {
           if (!chatController.getRunServiceInBackground()) SimplexService.stop(applicationContext)
         Lifecycle.Event.ON_START ->
           SimplexService.start(applicationContext)
+        Lifecycle.Event.ON_RESUME ->
```

Comment:
Will it fire when the app is first started? I agree that for the users that already have the app running we want to move it here, so it shows on resume, and not in onCreate... But maybe it should stay in onCreate as well?

Reply:
I am thinking to change the logic in such a way that while the app is in the foreground, the service will not be running, user will only manage the setting whether it should run, and then when the app is STOPPED (goes into background), the service will be started (if the setting is on).

But: can a user just kill the app without it ever going into background? In which case the service would never start, even if it is configured to be on? In which case it may be better to keep it on even while the app is in foreground? Or is there some event that would fire and can be used when the app is killed? And, would it start on timer if the app is killed (via scheduled worker)?...

Reply:
also, what I see is that ON_START and ON_RESUME always fire together...